lib/neovim-plugin.nix

```diff
@@ -79,6 +82,31 @@ with lib;
 
           package = helpers.mkPluginPackageOption originalName defaultPackage;
         }
+        // optionalAttrs (!skipConfigAttrs) {
+          config = {
+            pre = mkOption {
+              type = types.lines;
+              default = "";
+              description = "Configuration before the initialization of ${name}";
+            };
+            post = mkOption {
+              type = types.lines;
+              default = "";
+              description = "Configuration before the initialization of ${name}";
+            };
+            init = mkOption {
+              type = types.lines;
+              internal = true;
+              default = "";
+              description = "Initialization code (between pre & post)";
+            };
+            final = mkOption {
+              type = types.str;
+              description = "Generated lua configuration for ${name}";
+              readOnly = true;
+            };
+          };
```

Comment:
I think this would be better abstracted as a submodule type, that is a little more self-contained and re-usable:

```nix
pluginConfigText = types.submodule (
  { config, ... }:
  {
    options = {
      pre = mkOption {
        type = types.lines;
        default = "";
        description = "Configuration before the initialization of ${name}";
      };
      post = mkOption {
        type = types.lines;
        default = "";
        description = "Configuration before the initialization of ${name}";
      };
      init = mkOption {
        type = types.lines;
        internal = true;
        default = "";
        description = "Initialization code (between pre & post)";
      };
      final = mkOption {
        type = types.str;
        description = "Generated lua configuration for ${name}";
        readOnly = true;
      };
    };

    config.final = lib.nixvim.concatNonEmptyLines [
      config.pre
      config.init
      config.post
    ];
  });
```

Note the submodule can define `final` internally.

This could probably go in `nixvimTypes`?